rktlet/runtime/helpers.go

```diff
@@ -170,6 +171,34 @@ func getImageName(annotations map[string]string) string {
 	return name
 }
 
+func generateSeccompArg(annotations map[string]string, containerName string) string {
+	// by default kubernetes doesn't enable seccomp
+	defaultSeccomp := "--seccomp=mode=retain,@appc.io/all"
+
+	profile, ok := annotations[v1.SeccompContainerAnnotationKeyPrefix+containerName]
+	if !ok {
+		profile, ok = annotations[v1.SeccompPodAnnotationKey]
+		if !ok {
+			return defaultSeccomp
+		}
+	}
+
+	if profile == "unconfined" {
+		return defaultSeccomp
+	}
+
+	if profile == "docker/default" {
+		// use rkt default's, which matches docker's
+		return ""
+	}
+
+	// TODO(iaguis): handle custom profiles
+
+	// if we don't understand the profile, let's not enable seccomp to improve
+	// compatibility
+	return defaultSeccomp
```

Comment:
I feel that this goes in the opposite direction of what the user is trying to do. As far as I remember kubernetes will not pass this parameter by default, so can't we explicitly return an error telling the user that custom profiles are not supported?

Reply:
Right, it needs to be set explicitly so it's better to error out